spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/HttpClientProperties.java

```diff
@@ -157,6 +158,12 @@ public String toString() {
 		/** Only for type FIXED, the maximum time in millis to wait for aquiring. */
 		private Long acquireTimeout = ConnectionProvider.DEFAULT_POOL_ACQUIRE_TIMEOUT;
 
+		/**
+		 * Time in millis after which the channel will be closed,
+		 * if NULL there is no max idle time.
+		 */
+		private Long maxIdleTime = null;
```

Comment:
Please, set it to `Duration` instead of `Long` - then you will be able to use Spring Boot notation for resolving `Duration` from properties (https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-external-config-conversion-duration).